internal/service/project/resource_project_test.go

```diff
@@ -106,13 +106,23 @@ func TestGetProjectPropsFromAPI(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			svc := mocksvc.NewGroupProjectService(t)
-			svc.On("ListProjectTeams", mock.Anything, mock.Anything).Return(tc.teamRoleReponse.TeamRole, tc.teamRoleReponse.HTTPResponse, tc.teamRoleReponse.Err)
-			svc.On("ListProjectLimits", mock.Anything, mock.Anything).Return(tc.limitResponse.Limits, tc.limitResponse.HTTPResponse, tc.limitResponse.Err).Maybe()
-			svc.On("GetProjectSettings", mock.Anything, mock.Anything).Return(tc.groupResponse.GroupSettings, tc.groupResponse.HTTPResponse, tc.groupResponse.Err).Maybe()
-			svc.On("ReturnAllIPAddresses", mock.Anything, mock.Anything).Return(tc.ipAddressesResponse.IPAddresses, tc.ipAddressesResponse.HTTPResponse, tc.ipAddressesResponse.Err).Maybe()
+			teamsMock := mocksvc.NewTeamsApi(t)
+			projectsMock := mocksvc.NewProjectsApi(t)
```

Comment:
it sounds better to me to pass TeamsApi and ProjectsApi between functions instead of client :-)

Reply:
Changed in [b781b71](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2045/commits/b781b71f45f8db1b5555c70847644d57fc30763f)